test/jdk/java/io/File/EmptyPath.java

```diff
@@ -22,28 +22,278 @@
+    public void getFreeSpace() throws IOException {
+        FileStore fs = Files.getFileStore(f.toPath());
+        checkSpace(fs.getUnallocatedSpace(), f.getFreeSpace());
+    }
+
+    @Test
+    @Order(1)
+    public void getName() {
+        assertEquals(p.getFileName().toString(), f.getName());
+    }
+
+    @Test
+    @Order(1)
+    public void getParent() {
+        assertNull(f.getParent());
+    }
+
+    @Test
+    @Order(1)
+    public void getPath() {
+        assertEquals(p.toString(), f.getPath());
+    }
+
+    @Test
+    @Order(1)
+    public void getTotalSpace() throws IOException {
+        FileStore fs = Files.getFileStore(f.toPath());
+        checkSpace(fs.getTotalSpace(), f.getTotalSpace());
+    }
+
+    @Test
+    @Order(1)
+    public void getUsableSpace() throws IOException {
+        FileStore fs = Files.getFileStore(f.toPath());
+        checkSpace(fs.getUsableSpace(), f.getUsableSpace());
+    }
+
+    @Test
+    @Order(1)
+    public void isNotAbsolute() {
+        assertFalse(f.isAbsolute());
+    }
+
+    @Test
+    @Order(1)
+    public void isAbsolute() {
+        assertTrue(f.getAbsoluteFile().isAbsolute());
+    }
+
+    @Test
+    @Order(1)
+    public void isDirectory() {
+        assertTrue(f.isDirectory());
+    }
+
+    @Test
+    @Order(1)
+    public void isFile() {
+        assertFalse(f.isFile());
+    }
+
+    @Test
+    @Order(1)
+    public void isHidden() {
+        assertFalse(f.isHidden());
+    }
+
+    @Test
+    @Order(1)
+    public void lastModified() {
+        assertTrue(f.lastModified() > 0);
+    }
+
+    @Test
+    @Order(1)
+    public void length() throws IOException {
+        assertEquals(Files.size(f.toPath()), f.length());
+    }
+
+    @Test
+    @Order(1)
+    public void list() throws IOException {
+        String[] files = f.list();
+        assertNotNull(files);
+        Set<String> ioSet = new HashSet(Arrays.asList(files));
+        Set<String> nioSet = new HashSet();
+        Files.list(p).forEach((x) -> nioSet.add(x.toString()));
+        assertEquals(nioSet, ioSet);
+    }
+
+    @Test
+    @Order(1)
+    public void mkdir() {
+        assertFalse(f.mkdir());
+    }
+
+    @Test
+    @Order(2)
+    public void setLastModified() {
+        long t0 = f.lastModified();
+        long t = System.currentTimeMillis();
+        assertTrue(f.setLastModified(t));
+        assertEquals(t, f.lastModified());
+        assertTrue(f.setLastModified(t0));
+        assertEquals(t0, f.lastModified());
+    }
+
+    // Note: Testing File.setExecutable is omitted because calling
+    // File.setExecutable(false) makes it impossible to set the CWD to
+    // executable again which makes subsequent tests fail
+
+    @Test
+    @Order(3)
+    @DisabledOnOs({OS.WINDOWS})
+    public void setReadable() {
+        assertTrue(f.canRead());
+        assertTrue(f.setReadable(false));
+        assertFalse(f.canRead());
+        assertTrue(f.setReadable(true));
+        assertTrue(f.canRead());
```

Comment:
Need try-finally in the setter tests to ensure that the directory's permissions are restored in the event that the test fails.

Reply:
So changed in a3b7977.